controllers/control/cassandratask_controller.go

```diff
@@ -0,0 +1,559 @@
+	taskId := string(cassTask.UID)
+
+	for _, job := range cassTask.Spec.Jobs {
+		// TODO This should check the status of the job from Status, not reconciling it (we overwrite the jobId in those cases)
+		switch job.Command {
+		case "rebuild":
+			res, err = r.reconcileEveryPodTask(ctx, taskId, &dc, rebuild(job.Arguments))
+		case "cleanup":
+			res, err = r.reconcileEveryPodTask(ctx, taskId, &dc, cleanup())
+		case "decommission":
+			// res, err = r.reconcileEveryPodTask(ctx, taskId, &dc, decommission())
+		default:
+			err = fmt.Errorf("unknown job command: %s", job.Command)
+			return ctrl.Result{}, err
+		}
+
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if res.RequeueAfter > 0 {
+			// This job isn't complete yet or there's an error, do not continue
+			break
+		}
+		completedCount++
+	}
+
+	taskPatch := client.MergeFrom(cassTask.DeepCopy())
+
+	if cassTask.GetLabels() == nil {
+		cassTask.Labels = make(map[string]string)
+	}
+
+	if _, found := cassTask.GetLabels()[taskStatusLabel]; found {
+		if res.RequeueAfter == 0 && !res.Requeue {
+			// Job has been completed
+			cassTask.GetLabels()[taskStatusLabel] = completedTaskLabelValue
+			if err = r.Client.Patch(ctx, &cassTask, taskPatch); err != nil {
+				return res, err
+			}
+
+			cassTask.Status.Active = 0
+			cassTask.Status.CompletionTime = &timeNow
+
+			// Requeue for deletion later
+			deletionTime := calculateDeletionTime()
+			nextRunTime := deletionTime.Sub(timeNow.Time)
+			res.RequeueAfter = nextRunTime
+			logger.V(1).Info("This task has completed, scheduling for deletion in " + nextRunTime.String())
+		}
+	} else {
+		// Starting the run, set the Active label so we can quickly fetch the active ones
+		cassTask.GetLabels()[taskStatusLabel] = activeTaskLabelValue
+
+		if err = r.Client.Patch(ctx, &cassTask, taskPatch); err != nil {
+			return res, err
+		}
+
+		cassTask.Status.StartTime = &timeNow
+		cassTask.Status.Active = 1 // We don't have concurrency inside a task at the moment
+	}
+
+	if cassTask.Status.Succeeded != completedCount {
+		cassTask.Status.Succeeded = completedCount
+	}
+
+	// TODO Add conditions also
+	if err = r.Client.Status().Patch(ctx, &cassTask, taskPatch); err != nil {
+		return res, err
+	}
+
+	// log.V(1).Info for DEBUG logging
+	return res, err
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *CassandraTaskReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&api.CassandraTask{}).
+		Complete(r)
+}
+
+func (r *CassandraTaskReconciler) activeTasks(ctx context.Context, dc *cassapi.CassandraDatacenter) ([]api.CassandraTask, error) {
+	var taskList api.CassandraTaskList
+	matcher := client.MatchingLabels(utils.MergeMap(dc.GetDatacenterLabels(), map[string]string{taskStatusLabel: activeTaskLabelValue}))
+	if err := r.Client.List(ctx, &taskList, client.InNamespace(dc.Namespace), matcher); err != nil {
```

Comment:
This always returns an empty result set because there are no datacenter labels on a CassandraTask. You need to remove the datacenter labels and then I would filter the results  based on the `.Spec.Datacenter` field.